### ISSUE TITLE
Tracing issue

### ISSUE THREAD

**Issue by clyang82** [istio/istio#15993]
**Bug description**
cannot get the tracing data from jaeger after sending some requests to bookinfo. Here are my reproduce steps:
1. install the latest master build by using
```
helm install install/kubernetes/helm/istio --name istio --namespace istio-system --tls -f /tmp/istio-demo.yaml
``` 
istio-demo.yaml file content
```
gateways:
  istio-egressgateway:
    enabled: true
    resources:
      requests:
        cpu: 10m
        memory: 40Mi
  istio-ingressgateway:
    resources:
      requests:
        cpu: 10m
        memory: 40Mi
global:
  controlPlaneSecurityEnabled: false
  mtls:
    enabled: false
grafana:
  enabled: true
kiali:
  createDemoSecret: true
  enabled: true
mixer:
  adapters:
    stdio:
      enabled: true
  policy:
    enabled: true
    resources:
      requests:
        cpu: 10m
        memory: 100Mi
  telemetry:
    enabled: true
    resources:
      requests:
        cpu: 50m
        memory: 100Mi
pilot:
  resources:
    requests:
      cpu: 10m
      memory: 100Mi
  traceSampling: 100
sidecarInjectorWebhook:
  enabled: true
  rewriteAppHTTPProbe: false
tracing:
  enabled: true
```

Here we can see that `traceSampling: 100` which should collect all tracing data as sample. when i refresh the bookinfo example many times, there is no any data from jaeger. but when I send `for i in `seq 1 100`; do curl -s -o /dev/null http://9.30.167.192:31380/productpage; done`, there has one record generated and i can see it from jaeger.

When I redo the above steps with 1.2.2 official release, it is working. in other word, when I send a request to bookinfo, it generates one tracing record in jaeger.

I guess it should be broken by some fixing.

/cc @linsun @sdake  @howardjohn any insight?

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Comment by douglas-reid** [comment]
This should be being verified by existing e2e and integration tests. Do you perhaps have a cluster with out-of-sync clocks?

**Comment by clyang82** [comment]
@douglas-reid I just checked my environment. the clock is synced in my cluster. I still have such issue in my environment with istio-1.3.0-rc.1 it is easy to reproduce during my testing. Could you point me the e2e and integration test for this case? Thanks.

**Comment by howardjohn** [comment]
This was fixed in 1.3
